### PR TITLE
Category counts

### PR DESCRIPTION
**Pull request by az85252**
I have allowed the system to track the counts of each category rather than just a list. Like, instead of just listing the categories like [a,b,c], we can now internally have a way to count each category. The categories will return as before like [a,b,c], but inside, _categories variable is now something like [a:1,b:2,c:3].

### CODE REVIEW

File: dataprofiler/profilers/categorical_column_profile.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 from . import BaseColumnProfiler
 from .profiler_options import CategoricalOptions
 from . import utils
-
+from collections import Counter
+from collections import defaultdict
```
Comment (ChrisWallace2020) [inline dataprofiler/profilers/categorical_column_profile.py:5]: Can make this 1 import line with `from collections import Counter, defaultdict`, also, this is a standard library, so its import should go above the others

Comment (ChrisWallace2020) [reply dataprofiler/profilers/categorical_column_profile.py:5]: Also where are you using `Counter`? I don't see it anywhere in the code

File: requirements.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@ h5py>=2.10.0
 wheel>=0.33.1
 future>=0.18.2
 numpy>=1.18.5
+scipy>=1.4.1
```
Comment (ChrisWallace2020) [inline requirements.txt:5]: Why is this here?

**Comment by ChrisWallace2020** [comment]
Need to add some tests to ensure categorical data was counted as expected at initialization, update, and merge.

File: dataprofiler/profilers/categorical_column_profile.py (hunk 1 above)
Comment (JGSweets) [reply dataprofiler/profilers/categorical_column_profile.py:5]: I believe counter is not being used so we can remove it.

Comment (JGSweets) [reply dataprofiler/profilers/categorical_column_profile.py:5]: Also, need two spaces between imports and beginning of class.

File: requirements.txt (hunk 2 above)
Comment (JGSweets) [reply requirements.txt:5]: remove this req

**Comment by az85252** [comment]
Refer to Ability to internally store counts of each category in a CategoricalColumn in #296
